libcaf_net/caf/net/transport_worker_dispatcher.hpp

```diff
@@ -65,48 +68,48 @@ class transport_worker_dispatcher {
 
   template <class Parent>
   error handle_data(Parent& parent, span<byte> data, id_type id) {
-    auto it = workers_by_id_.find(id);
-    if (it == workers_by_id_.end()) {
-      // TODO: where to get node_id from here?
+    auto worker = find_by_id(id);
+    if (!worker) {
+      // TODO: where to get id_type from here?
       add_new_worker(parent, node_id{}, id);
-      it = workers_by_id_.find(id);
+      worker = find_by_id(id);
```

Comment:
We perform a lookup right after inserting the worker. `add_new_worker` should simply return the new worker.